### ISSUE TITLE
Trinity `ListDeserializationError` during fast sync

### ISSUE THREAD

**Issue by pipermerriam** [ethereum/py-evm#775]
### What is wrong?

```bash
concurrent.futures.process._RemoteTraceback:
"""
Traceback (most recent call last):
  File "/Users/piper/.pyenv/versions/3.6.3/lib/python3.6/concurrent/futures/process.py", line 175, in _process_worker
    r = call_item.fn(*call_item.args, **call_item.kwargs)
  File "/Users/piper/sites/py-evm/p2p/p2p_proto.py", line 56, in decode
    raw_decoded = cast(Dict[str, int], super(Disconnect, self).decode(data))
  File "/Users/piper/sites/py-evm/p2p/protocol.py", line 85, in decode
    return self.decode_payload(data[1:])
  File "/Users/piper/sites/py-evm/p2p/protocol.py", line 72, in decode_payload
    data = rlp.decode(rlp_data, sedes=decoder)
  File "/Users/piper/python-environments/trinity/lib/python3.6/site-packages/rlp/codec.py", line 220, in decode
    obj = sedes.deserialize(item, **kwargs)
  File "/Users/piper/python-environments/trinity/lib/python3.6/site-packages/eth_utils/functional.py", line 22, in inner
    return callback(fn(*args, **kwargs))
  File "/Users/piper/python-environments/trinity/lib/python3.6/site-packages/rlp/sedes/lists.py", line 84, in deserialize
    raise ListDeserializationError('List has wrong length', serial)
rlp.exceptions.ListDeserializationError: List has wrong length
"""

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Users/piper/sites/py-evm/p2p/peer.py", line 739, in connect
    self.cancel_token)
  File "/Users/piper/sites/py-evm/p2p/peer.py", line 143, in handshake
    await peer.do_p2p_handshake()
  File "/Users/piper/sites/py-evm/p2p/peer.py", line 222, in do_p2p_handshake
    cmd, msg = await self.read_msg()
  File "/Users/piper/sites/py-evm/p2p/peer.py", line 324, in read_msg
    token=self.cancel_token)
  File "/Users/piper/sites/py-evm/p2p/cancel_token.py", line 110, in wait_with_token
    return done.pop().result()
rlp.exceptions.ListDeserializationError: List has wrong length
```

This error occurs semi-regularly during fast sync.

### How can it be fixed

Explicit error handling.

**Comment by gsalgado** [comment]
This is one example of a message that causes this: 
```
b'\x01\xf8{\x05\xa9odash/v1.0.0-stable/windows-amd64/go1.8.3\xcc\xc5\x83eth?\xc5\x83eth>\x80\xb8@.\x1bb\x00s>\xb77W\xb6\x1bhK{V\x865\x8b\x0e\x15\x86\x90\xeb)\x90-\xee\xd0]\x95<\xac\xe2 \xc4\xf9\xf3\x01\xb4\x8a\xee\xdc\xea\x90b\xc4\t\xc9d\x18S$\xbb\xc9\xa1gI\x1clY\xe0\x94\xc4\xf9'
```

The first byte is the command id. In this case it is 0x01, which is the `Disconnect` command, but the payload (see below) matches exactly that of the `Hello` command (see https://github.com/ethereum/wiki/wiki/%C3%90%CE%9EVp2p-Wire-Protocol), which suggests a broken client implementation.

```
>>> rlp.decode(d[1:])
[b'\x05', b'odash/v1.0.0-stable/windows-amd64/go1.8.3', [[b'eth', b'?'], [b'eth', b'>']], b'', b'.\x1bb\x00s>\xb77W\xb6\x1bhK{V\x865\x8b\x0e\x15\x86\x90\xeb)\x90-\xee\xd0]\x95<\xac\xe2 \xc4\xf9\xf3\x01\xb4\x8a\xee\xdc\xea\x90b\xc4\t\xc9d\x18S$\xbb\xc9\xa1gI\x1clY\xe0\x94\xc4\xf9']
```

And we're not the first ones to notice that: https://github.com/ethereum/pyethapp/issues/288

I think the best alternative here is to catch that exception in `Disconnect.decode()` and log a warning

**Comment by cburgdorf** [comment]
Closing as addressed by #839
